src/nwbinspector/tools/_nwb.py

```diff
@@ -1,17 +1,20 @@
 """Helper functions related to NWB for internal use that rely on external dependencies (i.e., pynwb)."""
 
+from typing import Iterable, Type
+
 from pynwb import NWBFile
 
 
-def all_of_type(nwbfile: NWBFile, neurodata_type):
+def all_of_type(nwbfile: NWBFile, neurodata_type: Type) -> Iterable[object]:
     """Iterate over all objects inside an NWBFile object and return those that match the given neurodata_type."""
-    for obj in nwbfile.objects.values():
-        if isinstance(obj, neurodata_type):
-            yield obj
+    for neurodata_object in nwbfile.objects.values():
+        if isinstance(neurodata_object, neurodata_type):
+            yield neurodata_object
 
 
-def get_nwbfile_path_from_internal_object(obj):
+def get_nwbfile_path_from_internal_object(neurodata_object: object) -> str:
     """Determine the file path on disk for a NWBFile given only an internal object of that file."""
-    if isinstance(obj, NWBFile):
-        return obj.container_source
-    return obj.get_ancestor("NWBFile").container_source
+    if isinstance(neurodata_object, NWBFile):
+        return neurodata_object.container_source
+
+    return neurodata_object.get_ancestor("NWBFile").container_source  # type: ignore
```

Comment:
This type of ignore is a potentially interesting one - it's because I type a generic NWB-type object as a Python `object` so any of the special class methods like `.get_ancestor` naturally are more specific than that - so it would be cool to have a common base neurodata object in PyNWB that all types inherit from (does that already exist? For both datasets and their containers / groups?)

Reply:
(recommend making more specific typing improvements down the road, and when ready remove the ignores here)